File: pc01/pc01_04/c/pc01_04.c
/* lcd display
 * pc110104
 * uva706 */

#include <stdio.h>

#define MAX_NUMBER 99999999
#define MAX_SCALE 10

#define SEG_TOP        0b0000001
#define SEG_TOP_LEFT   0b0000010
#define SEG_TOP_RIGHT  0b0000100
#define SEG_MIDDLE     0b0001000
#define SEG_BOTTOM_LEFT  0b0010000
#define SEG_BOTTOM_RIGHT 0b0100000
#define SEG_BOTTOM     0b1000000

const int NUMBER[10] = {
	SEG_TOP | SEG_TOP_LEFT | SEG_TOP_RIGHT | SEG_BOTTOM_LEFT | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
	SEG_TOP_RIGHT | SEG_BOTTOM_RIGHT,
	SEG_TOP | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_LEFT | SEG_BOTTOM,
	SEG_TOP | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
	SEG_TOP_LEFT | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_RIGHT,
	SEG_TOP | SEG_TOP_LEFT | SEG_MIDDLE | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
	SEG_TOP | SEG_TOP_LEFT | SEG_MIDDLE | SEG_BOTTOM_LEFT | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
	SEG_TOP | SEG_TOP_RIGHT | SEG_BOTTOM_RIGHT,
	SEG_TOP | SEG_TOP_LEFT | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_LEFT | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
	SEG_TOP | SEG_TOP_LEFT | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_RIGHT | SEG_BOTTOM
};

void print_horizontal(int digit, int scale, int segment) {
	putchar(' ');
	for (int i = 0; i < scale; i++) {
		putchar((NUMBER[digit] & segment) ? '-' : ' ');
	}
	putchar(' ');
}

void print_vertical(int digit, int scale, int left_seg, int right_seg) {
	putchar((NUMBER[digit] & left_seg) ? '|' : ' ');
	for (int i = 0; i < scale; i++) {
		putchar(' ');
	}
	putchar((NUMBER[digit] & right_seg) ? '|' : ' ');
}

void print_digits(int number, int scale) {
	int digits[20];
	int length = 0;
	
	if (number == 0) {
		digits[length++] = 0;
	} else {
		while (number > 0) {
			digits[length++] = number % 10;
			number /= 10;
		}
	}
	
	for (int line = 0; line < 2 * scale + 3; line++) {
		for (int d = length-1; d >= 0; d--) {
			int digit = digits[d];
			
			if (line == 0) {
				print_horizontal(digit, scale, SEG_TOP);
			} 
			else if (line > 0 && line < scale + 1) {
				print_vertical(digit, scale, SEG_TOP_LEFT, SEG_TOP_RIGHT);
			}
			else if (line == scale + 1) {
				print_horizontal(digit, scale, SEG_MIDDLE);
			}
			else if (line > scale + 1 && line < 2 * scale + 2) {
				print_vertical(digit, scale, SEG_BOTTOM_LEFT, SEG_BOTTOM_RIGHT);
			}
			else if (line == 2 * scale + 2) {
				print_horizontal(digit, scale, SEG_BOTTOM);
			}
			
			if (d > 0) putchar(' ');
		}
		putchar('\n');
	}
}

int main(void) {
	int scale, number;
	
	while (1) {
		if (scanf("%d %d", &number, &scale) != 2 || (number == 0 && scale == 0)) {
			break;
		}

		if (number < 0 || number > (int) MAX_NUMBER) continue;
		if (scale < 1 || scale > MAX_SCALE) continue;
		
		print_digits(number, scale);
		putchar('\n');
	}
	
	return 0;
}

File: pc01/pc01_04/python/pc01_04.py
"""
lcd display
pc110104
uva706
"""

from typing import List
import sys

MAX_NUMBER = 99999999
MAX_SCALE = 10

# LCD segments as bit flags
SEG_TOP          = 0b0000001
SEG_TOP_LEFT     = 0b0000010
SEG_TOP_RIGHT    = 0b0000100
SEG_MIDDLE       = 0b0001000
SEG_BOTTOM_LEFT  = 0b0010000
SEG_BOTTOM_RIGHT = 0b0100000
SEG_BOTTOM       = 0b1000000

NUMBER = [
    SEG_TOP | SEG_TOP_LEFT | SEG_TOP_RIGHT | SEG_BOTTOM_LEFT | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
    SEG_TOP_RIGHT | SEG_BOTTOM_RIGHT,
    SEG_TOP | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_LEFT | SEG_BOTTOM,
    SEG_TOP | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
    SEG_TOP_LEFT | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_RIGHT,
    SEG_TOP | SEG_TOP_LEFT | SEG_MIDDLE | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
    SEG_TOP | SEG_TOP_LEFT | SEG_MIDDLE | SEG_BOTTOM_LEFT | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
    SEG_TOP | SEG_TOP_RIGHT | SEG_BOTTOM_RIGHT,
    SEG_TOP | SEG_TOP_LEFT | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_LEFT | SEG_BOTTOM_RIGHT | SEG_BOTTOM,
    SEG_TOP | SEG_TOP_LEFT | SEG_TOP_RIGHT | SEG_MIDDLE | SEG_BOTTOM_RIGHT | SEG_BOTTOM
]

def print_horizontal(digit: int, scale: int, segment: int) -> str:
    return ' ' + ('-' if NUMBER[digit] & segment else ' ') * scale + ' '

def print_vertical(digit: int, scale: int, left_seg: int, right_seg: int) -> str:
    return ('|' if NUMBER[digit] & left_seg else ' ') + ' ' * scale + ('|' if NUMBER[digit] & right_seg else ' ')

def get_digits(number: int) -> List[int]:
    if number == 0:
        return [0]
    
    digits = []
    while number > 0:
        digits.append(number % 10)
        number //= 10
    return digits[::-1]

def print_digits(number: int, scale: int) -> None:
    digits = get_digits(number)
    
    for line in range(2 * scale + 3):
        parts = []
        for digit in digits:
            if line == 0:
                parts.append(print_horizontal(digit, scale, SEG_TOP))
            elif 0 < line < scale + 1:
                parts.append(print_vertical(digit, scale, SEG_TOP_LEFT, SEG_TOP_RIGHT))
            elif line == scale + 1:
                parts.append(print_horizontal(digit, scale, SEG_MIDDLE))
            elif scale + 1 < line < 2 * scale + 2:
                parts.append(print_vertical(digit, scale, SEG_BOTTOM_LEFT, SEG_BOTTOM_RIGHT))
            elif line == 2 * scale + 2:
                parts.append(print_horizontal(digit, scale, SEG_BOTTOM))
        
        print(' '.join(parts))

def main():
    while True:
        try:
            scale, number = map(int, input().split())
            if scale == number == 0:
                break
            
            if not (0 <= number <= MAX_NUMBER and 1 <= scale <= MAX_SCALE):
                continue
                
            print_digits(number, scale)
            print()
            
        except EOFError:
            break

if __name__ == '__main__':
    main()